src/app/shared/spaces.service.ts

```diff
@@ -92,14 +102,26 @@ export class SpacesService implements Spaces {
     });
   }
 
-  private saveRecent(recent: Space[]) {
+  private saveRecent(recent: Space[]): Subscription {
     let patch = {
       store: {
         recentSpaces: recent.map(val => val.id)
       }
     } as ExtProfile;
     return this.profileService.silentSave(patch)
-      .subscribe(profile => { }, err => console.log('Error saving recent spaces:', err));
+      .subscribe(() => {}, err => console.log('Error saving recent spaces:', err));
+  }
+
+  /**
+   * Given the id of a space, returns the index of the space in _recent
+   */
+  findRecentIndexById(spaces: Space[], id: string): number {
+    for (let i: number = 0; i < spaces.length; i++) {
+      if (spaces[i].id === id) {
+        return i;
+      }
+    }
+    return -1;
```

Comment:
This looks a lot like findIndex: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

`return spaces.findIndex(space => space.id === id)`

Reply:
Ah thanks for the heads up, that does look like what I want.